File: leetcode/0014. Longest Common Prefix/Solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0 or len(strs[0]) == 0:
            return ""

        result = ""
        for i in range(len(strs[0])):
            current_char = strs[0][i]
            for word in strs:
                if i == len(word) or word[i] != current_char:
                    return result
            result += current_char
        return result

File: leetcode/0014. Longest Common Prefix/Solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder result = new StringBuilder();
        int currentIndex = 0;
        while(true) {
            if (strs.length == 0 || strs[0].length() == currentIndex) {
                return result.toString();
            }
            char currentChar = strs[0].charAt(currentIndex);
            for(int i = 1; i < strs.length; i++) {
                if (strs[i].length() == currentIndex ||
                      strs[i].charAt(currentIndex) != currentChar) {
                    return result.toString();
                }
            }
            result.append(currentChar);
            currentIndex++;
        }
    }
}
